### PR TITLE
Revisit executable sizes, and make nonroot standard and not an option

### PR DESCRIPTION
**Pull request by palladia**
1. Omiserver reduced to 1/4 of its earlier size
2. Omiengine no longer auto-restarted
3. Omeagent auto-shutdowns if omiserver dies.

### CODE REVIEW

File: Unix/server/GNUmakefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ INCLUDES = $(TOP) $(TOP)/common
 
 DEFINES = HOOK_BUILD MI_CONST=
 
-LIBRARIES = wsman xmlserializer http protocol sock disp provreg provmgr $(MIAPILIB) wql base xml $(PALLIBS) $(INDIMGRLIB) omi_error $(INDICMNLIB) micodec mofparser
+LIBRARIES = protocol sock $(MIAPILIB) wql base $(PALLIBS) $(INDIMGRLIB) omi_error $(INDICMNLIB) micodec mofparser
```
Comment (paulcallen) [inline Unix/server/GNUmakefile]: other things that should be able to go include:
* miapi, wql, micodec, mofbase
things I would also expect to be able to go include:
* indicationmgr and omi_error

File: Unix/agent/agent.c [hunk 2]
```diff
@@ -422,10 +425,20 @@ int agent_main(int argc, const char* argv[])
...
     /* Run the protocol object (waiting for new messages) */
-    r = Protocol_Run( &s_data.protocol->internalProtocolBase, TIME_NEVER);
+    for (;;)
+    {
+        r = Protocol_Run( &s_data.protocol->internalProtocolBase, ONE_SECOND_USEC);
 
-    if (r != MI_RESULT_OK)
-        err(ZT("Protocol_Run() failed (%d)"), (int)r);
+        if (r == MI_RESULT_FAILED)
+            err(ZT("Protocol_Run() failed (%d)"), (int)r);
```
[the remark this answers is not in the document]
Comment (palladia) [reply Unix/agent/agent.c]: err() calls exit(1)

File: Unix/server/GNUmakefile (hunk 1 above)
Comment (palladia) [reply Unix/server/GNUmakefile]: Removed

File: Unix/server/server.c [hunk 3]
```diff
@@ -62,7 +61,7 @@ static int _StartEngine(int argc, char** argv, char ** envp, const char *engineS
     Strlcat(engineFile, "/omiengine", PAL_MAX_PATH_SIZE);
     argv[0] = engineFile;
 
-    r = BinaryProtocolListenFile(engineSockFile, &s_data.mux[0], &s_data.protocol0, secretString);
+    r = BinaryProtocolListenFile(engineSockFile, NULL, &s_data.protocol0, secretString);
```
Comment (paulcallen) [inline Unix/server/server.c]: I finally understand why this is possible as it means you are put handling the new non-root messages within server itself, rather in the protocol layer itself. This needs to be put back, the code relocated into the server and basically handle strands properly.

**Comment by palladia** [comment]
+microsoft/omi-devs

**Comment by palladia** [comment]
@microsoft/omi-devs

**Comment by palladia** [comment]
- Moved some code from protocol to server/engine

- Rebased with release 1.4

@microsoft/omi-devs  Please review

**Comment by palladia** [comment]
OMI now in maintenance mode.  Changes archived indefinitely.

The 4th commit of this PR has been cherry-picked into its own PR.

**Pull request closed without merging**